eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -232,6 +235,31 @@ object PaymentFailure {
 
 }
 
+case class PaymentFailureSummary(amount: MilliSatoshi, route: Seq[PublicKey], message: String)
+
+object PaymentFailureSummary {
+  def apply(failure: PaymentFailure): PaymentFailureSummary = {
+    val route = failure.route.map(_.nodeId) ++ failure.route.lastOption.map(_.nextNodeId)
+    val message = failure match {
+      case LocalFailure(_, _, t) => t.getMessage
+      case RemoteFailure(_, _, Sphinx.DecryptedFailurePacket(origin, failureMessage)) => s"$origin returned: ${failureMessage.message}"
+      case _: UnreadableRemoteFailure => "unreadable remote failure"
+    }
+    PaymentFailureSummary(failure.amount, route, message)
+  }
+}
+
+case class PaymentFailedSummary(totalAmount: MilliSatoshi, pathFindingExperiment: String, failures: Seq[PaymentFailureSummary]) {
+  def json(): String = {
+    import fr.acinq.eclair.json.JsonSerializers
+    JsonSerializers.serialization.write(this)(JsonSerializers.formats)
```

Comment:
This needs to be in a `Try` (json serialization errors do happen):

```suggestion
    import scala.util.Try
    Try(JsonSerializers.serialization.write(this)(JsonSerializers.formats)).getOrElse("json serialization error")
```

(or make the method return the `Try` and handle the error above)

Reply:
Done